neqo-http3-server/src/main.rs

```diff
@@ -214,6 +238,7 @@ fn process(
     }
 }
 
+#[allow(clippy::cognitive_complexity)]
```

Comment:
This function does many things:

1. setup (parse arguments, init_db)
2. build servers
3. process input from servers

That seems like at least two functions that can be extracted.  It looks like the socket setup can be extracted too.